### PR TITLE
Clean-up LoRA flow

### PR DESCRIPTION
**Pull request by SanjuCSudhakaran**
This PR cleans up LoRA flow by removing unnecessary functions and variables.

Removed special handling of `max_num_batched_tokens` for HPU in `models.py` since we internally handle this in  PunicaWrapperHPU [PR](https://github.com/vllm-project/vllm/blob/d51d66c3252107d5b986d2eab7af1c210dceb708/vllm/lora/punica_wrapper/punica_hpu.py#L17)

Removed `convert_mapping` from `models.py` based on this [PR](https://github.com/vllm-project/vllm/pull/5036/files#:~:text=def%20convert_mapping)

### CODE REVIEW

File: vllm/worker/hpu_model_runner.py [hunk 1]
```diff
@@ -1904,15 +1904,6 @@ def get_counter_dict(self, cache_config, duration, seq_len,
         return counters
 
 
-def unwrap_model(model):
```
[the remark this answers is not in the document]
Comment (SanjuCSudhakaran) [reply vllm/worker/hpu_model_runner.py:1907]: Yes, we can remove this function.

These changes were added to handle `lora_logits` in compile mode. https://github.com/HabanaAI/vllm-fork/pull/196

Since we moved the special handling of logits to `layers.py` this function is no longer required.  https://github.com/HabanaAI/vllm-fork/pull/339/files#diff-7f24556108f1fe144f66e06dabdb5c1c547431ae0481112c8194d5e5d76ae996R1206

**Review by vivekgoe — APPROVED** [review]
